opennebula/resource_opennebula_virtual_machine.go

```diff
@@ -1261,15 +1261,79 @@ func resourceOpennebulaVirtualMachineUpdate(d *schema.ResourceData, meta interfa
 	}
 
 	updateConf := false
+
+	// retrieve only template sections managed by updateconf method
 	tpl := vm.NewTemplate()
+	for _, name := range []string{"OS", "FEATURES", "INPUT", "GRAPHICS", "RAW", "CONTEXT"} {
```

Comment:
why in the following lines you work only on "OS", "GRAPHICS" and "CONTEXT".
Why not "FEATURES", "INPUT" or "RAW" ?

Reply:
I copy these sections only, from the existing VM template, because the method updateconf take them into account:
https://docs.opennebula.io/5.12/integration/system_interfaces/api.html#one-vm-updateconf

If I don't copy one of these fields, it would be erased. See `OS` deletion problem when updating CONTEXT in #169 .
The RAW field is managed through the template resource, I didn't enabled it in VM resource, here is the original issue about this section: https://github.com/OpenNebula/terraform-provider-opennebula/issues/123
Do we need it in VM resource ?

For other parts FEATURES and INPUT there aren't managed at all.
This is not the goal of this issue, we could open a new issue (probably better) or just add them to https://github.com/OpenNebula/terraform-provider-opennebula/pull/154

Reply:
perfect thx